hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -740,6 +746,18 @@ public void stop() {
     ozoneManagerDoubleBuffer.stop();
     HadoopExecutors.shutdown(executorService, LOG, 5, TimeUnit.SECONDS);
     HadoopExecutors.shutdown(installSnapshotExecutor, LOG, 5, TimeUnit.SECONDS);
+    LOG.info("applyTransactionMap size {} ", applyTransactionMap.size());
```

Comment:
As discussed in yesterday's APAC community sync, it would be nice to make this size a metric.

Reply:
> As discussed in yesterday's APAC community sync, it would be nice to make this size a metric.

Metric is useful. May be we can update metric when `applyTransactionMap.put/remove` and `ratisTransactionMap.put/remove`.  I added a metric in my env and validated monitoring values:
```js
{
    "name" : "Hadoop:service=OzoneManager,name=OMPerformanceMetrics",
    "modelerType" : "OMPerformanceMetrics",
    "tag.Hostname" : "bigdata-ozonemaster00.xxx.com",
    "ApplyTransactionMapSize" : 49,  // expect 0 when writing done
    //...
    "RatisTransactionMapSize" : 25,  // expect 0 when writing done
    //...
  }, {
```
The applyTransactionMap and ratisTransactionMap trace as follow:
```java
2023-12-10 00:41:26,939 [om0@group-88932ED033FB-StateMachineUpdater] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():1, perfMetrics.applyTransactionMapSize: 1, ratisTransactionMap.size(): 0, perfMetrics.ratisTransactionMapSize: 0
2023-12-10 00:41:26,947 [om0@group-88932ED033FB-StateMachineUpdater] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():2, perfMetrics.applyTransactionMapSize: 2, ratisTransactionMap.size(): 0, perfMetrics.ratisTransactionMapSize: 0
...
2023-12-10 00:41:26,959 [om0@group-88932ED033FB-StateMachineUpdater] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():49, perfMetrics.applyTransactionMapSize: 49, ratisTransactionMap.size(): 0, perfMetrics.ratisTransactionMapSize: 0
2023-12-10 00:41:26,959 [om0@group-88932ED033FB-StateMachineUpdater] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():50, perfMetrics.applyTransactionMapSize: 50, ratisTransactionMap.size(): 0, perfMetrics.ratisTransactionMapSize: 0
2023-12-10 00:41:26,994 [OMDoubleBufferFlushThread] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():49, perfMetrics.applyTransactionMapSize: 49, ratisTransactionMap.size(): 25, perfMetrics.ratisTransactionMapSize: 25
2023-12-10 00:41:27,001 [OMDoubleBufferFlushThread] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():48, perfMetrics.applyTransactionMapSize: 48, ratisTransactionMap.size(): 25, perfMetrics.ratisTransactionMapSize: 25
...
2023-12-10 00:41:27,064 [OMDoubleBufferFlushThread] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():0, perfMetrics.applyTransactionMapSize: 0, ratisTransactionMap.size(): 2, perfMetrics.ratisTransactionMapSize: 2
2023-12-10 00:41:27,064 [OMDoubleBufferFlushThread] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():0, perfMetrics.applyTransactionMapSize: 0, ratisTransactionMap.size(): 1, perfMetrics.ratisTransactionMapSize: 1
2023-12-10 00:41:27,064 [OMDoubleBufferFlushThread] DEBUG org.apache.hadoop.ozone.om.ratis.OzoneManagerStateMachine: applyTransactionMap.size():0, perfMetrics.applyTransactionMapSize: 0, ratisTransactionMap.size(): 0, perfMetrics.ratisTransactionMapSize: 0
```